File: backend-java/src/main/java/com/jerrygram/application/dtos/PostListItemDto.java
package com.jerrygram.application.dtos;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.jerrygram.infrastructure.configuration.UuidDeserializer;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PostListItemDto {
    @JsonDeserialize(using = UuidDeserializer.class)
    private UUID id;
    private String caption;
    private String imageUrl;
    
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", timezone = "UTC")
    private LocalDateTime createdAt;
    
    private int likes;
    private boolean liked;
    private SimpleUserDto user;
    private double score;
}

File: backend-dotnet/Application/DTOs/PostListItemDto.cs
﻿namespace Application.DTOs
{
    public class PostListItemDto
    {
        public Guid Id { get; set; }
        public string? Caption { get; set; }
        public string? ImageUrl { get; set; }
        public DateTime CreatedAt { get; set; }
        public int Likes { get; set; }
        public bool Liked { get; set; }
        public SimpleUserDto User { get; set; } = null!;
        public double Score { get; set; }
    }
}
